### PR TITLE
Faucet fix - sends tx successfully

### PR DESCRIPTION
**Pull request by araskachoi**
Closes: https://github.com/ChainSafe/ethermint/issues/522

## Description

Include logic to successfully send a transaction from the faucet account to a recipient account. This must be done on the node that owns the faucet account.

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: x/faucet/client/cli/tx.go [hunk 1]
```diff
@@ -61,39 +61,62 @@ func GetCmdRequest(cdc *codec.Codec) *cobra.Command {
...
-			var recipient sdk.AccAddress
-			if len(args) == 1 {
-				recipient = cliCtx.GetFromAddress()
+			var faucet, recipient sdk.AccAddress
+
+			// accepts bech32 or acct name for faucet
+			if len(args[0]) < 42 {
+				for _, info := range infos {
+					if args[0] == info.GetName() {
+						faucet = info.GetAddress()
```
Comment (fedekunze) [inline x/faucet/client/cli/tx.go:75]: break after?

File: x/faucet/client/cli/tx.go [hunk 2]
```diff
@@ -61,39 +61,62 @@ func GetCmdRequest(cdc *codec.Codec) *cobra.Command {
...
 
-			if err != nil {
-				fmt.Println("failed to create acc address")
-				return err
+			// accepts bech32 or acct name for recipient
+			if len(args[1]) < 42 {
+				for _, info := range infos {
+					if args[1] == info.GetName() {
+						recipient = info.GetAddress()
+					}
```
Comment (fedekunze) [inline x/faucet/client/cli/tx.go]: we should use a util function, afaik this is used in other parts too

File: x/faucet/client/cli/tx.go [hunk 3]
```diff
@@ -61,39 +61,62 @@ func GetCmdRequest(cdc *codec.Codec) *cobra.Command {
...
+			if err := msg.ValidateBasic(); err != nil {
+				return err
 			}
 
 			err = authclient.GenerateOrBroadcastMsgs(cliCtx, txBldr, []sdk.Msg{msg})
 			if err != nil {
 				fmt.Println("failed to run request command:", err)
 			}
+
 			return err
```
Comment (fedekunze) [inline x/faucet/client/cli/tx.go]:
```suggestion
			return authclient.GenerateOrBroadcastMsgs(cliCtx, txBldr, []sdk.Msg{msg})
```

File: x/faucet/client/cli/tx.go (hunk 2 above)
Comment (araskachoi) [reply x/faucet/client/cli/tx.go]: i see it used as an unexported fn in the cosmos-sdk methods. any pointers on where we use it in our code (or our implementation of that util method)?

**Review by fedekunze — COMMENTED** [review]
I still get the error 

```go
ethermintcli tx faucet request $(ethermintcli keys show other -a) $(ethermintcli keys show other -a) 100aphoton
ERROR: faucet account does not exist: eth179pz7vdxzjeyqw30rrnz2trl86kghcntnzcpul
```

**Comment by araskachoi** [comment]
> I still get the error
> 
> ```go
> ethermintcli tx faucet request $(ethermintcli keys show other -a) $(ethermintcli keys show other -a) 100aphoton
> ERROR: faucet account does not exist: eth179pz7vdxzjeyqw30rrnz2trl86kghcntnzcpul
> ```

the command should be `request [faucet-address] [recipient-address] [amount]`
so should be:
```go
ethermintcli tx faucet request $(ethermintcli keys show mykey -a) $(ethermintcli keys show other -a) 100aphoton
```

**Comment by fedekunze** [comment]
hmm isn't that the same behaviour as before tho? 

```
ethermintcli tx faucet 100photons $(ethermintcli keys show other -a) --from mykey
```

**Comment by araskachoi** [comment]
yeap pretty much! except it doesnt take the `from` address from context (in case you have multiple accounts on your node). so i think it's better to be explicit about which faucet account you are requesting from as well as the recipient of the token

**Pull request closed without merging**